### ISSUE TITLE
truffle console deployed contract acting strange

### ISSUE THREAD

**Issue by sajadghawami** [trufflesuite/truffle#371]
Hey,

## Issue
after compiling and migrating a contract named ```Proof```, i do a ```truffle console```.

This is what happens:

```
truffle(development)> var test = Proof.deployed()
Using network 'development'.
  0 passing (0ms)

truffle(development)> var test2 = Proof.deployed()
undefined

truffle(development)> var test = ProofXX.deployed()
Using network 'development'.
  0 passing (1ms)

````

## Steps to Reproduce

Fresh ```truffle init``` on macOS

## Environment

* Operating System:  macOS sierra
* Truffle version: 3.1.2
* node version: 7.5.0
* npm version: 4.1.2

**Comment by predicador37** [comment]
Why did you close the issue? I'm having similar problems. @sajadghawami could you solve your problems?

**Comment by tcoulter** [comment]
The behavior seen by @sajadghawami is due to a recent bug we introduced this week. See here: https://github.com/trufflesuite/truffle-core/pull/29

We'll have a fix out Monday.

**Comment by predicador37** [comment]
Thanks a lot @tcoulter . I'll take another look by Monday then.

**Comment by sajadghawami** [comment]
@predicador37 sorry, its already a while back and unfortunately i don't remember what i did to solve this issue...

**Comment by predicador37** [comment]
The problem is solved in the last release, 3.4.7.
Thanks a lot!

**Comment by tcoulter** [comment]
Closing! Cheers!
